src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -4720,4 +4889,1195 @@ public class C
+            // x = c.Y1;
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "c.Y1").WithArguments($"{modifier} int p", "ref readonly int p").WithLocation(9, 5),
+            // (10,5): warning CS9510: Reference kind modifier of parameter 'ref readonly int p' doesn't match the corresponding parameter 'in int p' in target.
+            // y = c.X1;
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "c.X1").WithArguments("ref readonly int p", $"{modifier} int p").WithLocation(10, 5),
+            // (15,5): warning CS9510: Reference kind modifier of parameter 'in int p' doesn't match the corresponding parameter 'ref readonly int p' in target.
+            // x = c.Y2;
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "c.Y2").WithArguments($"{modifier} int p", "ref readonly int p").WithLocation(15, 5),
+            // (16,5): warning CS9510: Reference kind modifier of parameter 'ref readonly int p' doesn't match the corresponding parameter 'in int p' in target.
+            // y = c.X2;
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "c.X2").WithArguments("ref readonly int p", $"{modifier} int p").WithLocation(16, 5));
+    }
+
+    [Theory, CombinatorialData]
+    public void Conversion_Tuple([CombinatorialValues("in", "ref")] string modifier)
+    {
+        var source = $$"""
+            (X x, Y y) = (C.X, C.Y);
+            
+            var i = 1;
+            x(in i);
+            y({{modifier}} i);
+            
+            (x, y) = (C.Y, C.X);
+            
+            x(in i);
+            y({{modifier}} i);
+            
+            class C
+            {
+                public static void X(ref readonly int p) => System.Console.Write("X");
+                public static void Y({{modifier}} int p) => System.Console.Write("Y");
+            }
+            
+            delegate void X(ref readonly int p);
+            delegate void Y({{modifier}} int p);
+            """;
+        CompileAndVerify(source, expectedOutput: "XYYX").VerifyDiagnostics(
+            // (7,11): warning CS9510: Reference kind modifier of parameter 'in int p' doesn't match the corresponding parameter 'ref readonly int p' in target.
+            // (x, y) = (C.Y, C.X);
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "C.Y").WithArguments($"{modifier} int p", "ref readonly int p").WithLocation(7, 11),
+            // (7,16): warning CS9510: Reference kind modifier of parameter 'ref readonly int p' doesn't match the corresponding parameter 'in int p' in target.
+            // (x, y) = (C.Y, C.X);
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "C.X").WithArguments("ref readonly int p", $"{modifier} int p").WithLocation(7, 16));
+    }
+
+    [Theory, CombinatorialData]
+    public void Conversion_Nullable([CombinatorialValues("in", "ref")] string modifier)
```

Comment:
>Conversion_Nullable [](http://example.com/codeflow?start=16&length=19)

Is this test meaningful? Delegate types are reference types, ```X?``` doesn't mean ```Nullable<X>```, therefore, I do not think any nullable conversion is involved here.

Reply:
Good catch, haven't realized that.